### ISSUE TITLE
Deep Learning Skills

### ISSUE THREAD

**Issue by jasonfresh** [jasonfresh/notes#7]
### Batch Normalization
- [A Gentle Introduction to Batch Normalization for Deep Neural Networks](https://machinelearningmastery.com/batch-normalization-for-training-of-deep-neural-networks/)
    - lagre batch is better: For small mini-batch sizes or mini-batches that do not contain a representative distribution of examples from the training dataset, the differences in the standardized inputs between training and inference (using the model after training) can result in noticeable differences in performance. 
    - Batch normalization may be used on the inputs to the layer before or after the activation function in the previous layer.
        - more appropriate after the activation function if for s-shaped functions like the hyperbolic tangent and logistic function.
        - It may be appropriate before the activation function for activations that may result in non-Gaussian distributions like the rectified linear activation function, the modern default for most network types.

- Why did you freeze the BN when training ?
    -  It's a common practice to freeze BN while training object detection models. This is done because you usually start from some Imagenet pre-trained weights which were trained with large BS (like 256 or larger) while in object detection BS is much smaller. You don't want to ruin good pretrained statistics so freezing BN is a good idea

**Comment by jasonfresh** [comment]
### Ref
- [Adam: A Method for Stochastic Optimization (slides)](https://moodle2.cs.huji.ac.il/nu15/pluginfile.php/316969/mod_resource/content/1/adam_pres.pdf)

**Comment by jasonfresh** [comment]
- [A Disciplined Approach To Neural Network Hyper-Parameters: Part I - Learning Rate, Batch Size, Momentum, And Weight Decay](https://arxiv.org/pdf/1803.09820.pdf)
